lib/3scale_toolbox/commands/import_command/openapi/resource_reader.rb

```diff
@@ -0,0 +1,50 @@
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        module ResourceReader
+          ##
+          # Reads resources from different types of sources.
+          # Supported types are: file, URL, stdin
+          # Return type is
+          # [content, format] where
+          # content: raw content
+          # format: Hash with single key: `format`. Value can be `:json` or `:yaml`
+          # format example: { format: :json }
+          def openapi_resource(resource)
+            case resource
+            when '-'
+              method(:read_stdin)
+            when /\A#{URI::DEFAULT_PARSER.make_regexp}\z/
+              method(:read_url)
+            else
+              method(:read_file)
+            end.call(resource)
+          end
+
+          # Detect format from file extension
+          def read_file(resource)
+            [File.read(resource), { format: File.extname(resource) }]
+          end
+
+          def read_stdin(_resource)
+            content = STDIN.read
+            # will try parse json, otherwise yaml
+            format = :json
+            begin
+              JSON.parse(content)
```

Comment:
JSON is a subset of YAML, so parsing it as YAML should work.

Reply:
I would not say JSON is a subset of YAML :)

Anyway, it is true, indeed, that yaml parser in ruby parses json syntax.

The swagger parser interface requires two params, content stream and format (`json` or `yaml`). `read_stdin` method has no way to find out format other than checking if json parser fails. The result is not even used. So we need to parse to find out format.

Or we can always tell swagger parser it is yaml, even when it is json. But I guess it is ugly and haven't checked if it works. But the interface makes clear distinction between `json` nad `yaml`

Reply:
YAML is a superset of JSON. Not only in Ruby. Everywhere.

Yes, you can use YAML just fine https://github.com/swagger-rb/swagger-rb/blob/51b0a68e5c24e9b8eeefb3467700da58d5cac884/lib/swagger/parsers.rb#L26.

You can remove almost everyting, use ARGF to read all files and stdin, always use YAML and just detect if there is an URL you need.

Reply:
And you can just pass already parsed document to the swagger library: https://github.com/swagger-rb/swagger-rb/blob/51b0a68e5c24e9b8eeefb3467700da58d5cac884/lib/swagger.rb#L25

That makes more sense anyway to separate concerns of parsing/loading and the actual spec.

Reply:
Using `YAML.load` and removing format tracking login
 https://github.com/3scale/3scale_toolbox/pull/76/commits/8bba8577d2b1b76dfa3a823537dab9abe8bddd8a